tests/punica.test.js

```diff
@@ -599,6 +629,155 @@ describe('Rest API interface', () => {
       });
     });
 
+    describe('getDevicesEntries function', () => {
+      it('should return a list of registered devices entries', () => {
+        nock(url)
+          .get('/devices')
+          .reply(200, response.devicesEntries);
+        return service.getDevicesEntries().then((resp) => {
+          expect(typeof resp).to.equal('object');
+          expect(resp[0]).to.have.property('psk_id');
+          expect(resp[0]).to.have.property('uuid');
+        });
+      });
+
+      it('should return an error (status code number) if status code is not 200', () => {
+        nock(url)
+          .get('/devices')
+          .reply(500);
+        return service.getDevicesEntries().catch((err) => {
+          expect(typeof err).to.equal('number');
+        });
+      });
+
+      it('should return rejected promise with exception object if connection is not succesfull', (done) => {
+        service.getDevicesEntries()
+          .catch((err) => {
+            expect(typeof err).to.equal('object');
+            done();
+          });
+      });
+    });
+
+    describe('getDeviceEntry function', () => {
+      const name = 'threeSeven';
+      it('should return device\'s entry', () => {
+        nock(url)
+          .get(`/devices/${name}`)
+          .reply(200, response.entry);
+        return service.getDeviceEntry(name).then((resp) => {
+          expect(typeof resp).to.equal('object');
+          expect(resp).to.have.property('psk_id');
+          expect(resp).to.have.property('uuid');
+        });
+      });
+
+      it('should return an error (status code number) if status code is not 200', () => {
+        nock(url)
+          .get(`/devices/${name}`)
+          .reply(500);
+        return service.getDeviceEntry(name).catch((err) => {
+          expect(typeof err).to.equal('number');
+        });
+      });
+
+      it('should return rejected promise with exception object if connection is not succesfull', (done) => {
+        service.getDeviceEntry(name)
+          .catch((err) => {
+            expect(typeof err).to.equal('object');
+            done();
+          });
+      });
+    });
+
+    describe('registerDeviceEntry function', () => {
+      const data = { psk: 'cHNrMQ==', psk_id: 'cHNraWQx' };
+      it('should fulfill promise if registration is successfull (201)', () => {
```

Comment:
There have been some changes to the API lately. Most importantly, UUID is generated by the server and is returned in the response of the POST request. The current punica API documentation does not reflect this, and will be fixed soon. You later can use the returned UUID during tests where you edit/delete/get a specific device entry.